### PR TITLE
add all project

### PR DESCRIPTION
**Pull request by alex-slapoguzov**

### CODE REVIEW

File: src/test/java/CartTests.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import parser.JsonParser;
+import parser.Parser;
+import shop.Cart;
+import shop.RealItem;
+import shop.VirtualItem;
+
+import java.io.File;
+
+
+public class CartTests {
```
Comment (EugenBorisik) [inline src/test/java/CartTests.java]: Rename into CartTest. Do not use plurals in namings

File: src/test/java/CartTests.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    private Cart cart;
+    private RealItem realItem;
+    private VirtualItem virtualItem;
+    private Parser parser;
+
```
Comment (EugenBorisik) [inline src/test/java/CartTests.java]: Remove 1 line

File: src/test/java/CartTests.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    @ParameterizedTest
+    @CsvSource({"src\\test\\resources\\andrew-cart.json, 38445.48", "src\\test\\resources\\eugen-cart.json, 26560.68"})
+    void getTotalPriceWithParserClassPositiveTest(String pathName, double totalPrice) {
+        parser = new JsonParser();
+        cart = parser.readFromFile(new File(pathName));
+        Assertions.assertEquals(totalPrice, cart.getTotalPrice(), 0.1);
+    }
+
```
Comment (EugenBorisik) [inline src/test/java/CartTests.java]: Remove 1 line

File: src/test/java/CartTests.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    @ParameterizedTest
+    @CsvFileSource(resources = "\\index.csv")
+    void getTotalPriceWithDataPositiveTest(double priceRealItem, double priceVirtualItem, double totalPrice) {
+
```
Comment (EugenBorisik) [inline src/test/java/CartTests.java]: Remove empty line

File: src/test/java/CartTests.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        cart = new Cart("igor-cart");
+
+        realItem = new RealItem();
+        virtualItem = new VirtualItem();
+
```
Comment (EugenBorisik) [inline src/test/java/CartTests.java]: Remove 1 line

File: src/test/java/CartTests.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+        realItem.setPrice(priceRealItem);
+        virtualItem.setPrice(priceVirtualItem);
+
```
Comment (EugenBorisik) [inline src/test/java/CartTests.java]: Remove 1 line

File: src/test/java/JsonParserTests.java [hunk 7]
```diff
@@ -0,0 +1,131 @@
+import org.junit.jupiter.api.*;
```
Comment (EugenBorisik) [inline src/test/java/JsonParserTests.java]: Avoid asterisks in imports

File: src/test/java/JsonParserTests.java [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+import org.junit.jupiter.params.provider.ValueSource;
+import parser.JsonParser;
+import parser.Parser;
+import shop.Cart;
+
+
+import java.io.File;
+
+
+public class JsonParserTests {
```
Comment (EugenBorisik) [inline src/test/java/JsonParserTests.java]: Rename into JsonParserTest. Do not use plurals in namings

File: src/test/java/JsonParserTests.java [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+
```
Comment (EugenBorisik) [inline src/test/java/JsonParserTests.java]: Remove 1 line

File: src/test/java/JsonParserTests.java [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+
+    private Parser jsonParser;
+    private Cart cart;
+
```
Comment (EugenBorisik) [inline src/test/java/JsonParserTests.java]: Remove 1 line

File: src/test/java/JsonParserTests.java [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+    @BeforeEach
+    void setUp() {
+        jsonParser = new JsonParser();
+    }
+
+    @AfterEach
+    void tearDown() {
+        String pathName = "src/main/resources/";
+
+        File file = new File(pathName);
```
Comment (EugenBorisik) [inline src/test/java/JsonParserTests.java]: Take a look on https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/FileUtils.html
I recommend to use it

File: src/test/java/JsonParserTests.java [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+            if (!(item.getName().equals("andrew-cart.json")
+                    | item.getName().equals("eugen-cart.json"))) {
+
+                item.delete();
+            }
+        }
+    }
+
+    @Nested
+    class writeToFileTests {
```
Comment (EugenBorisik) [inline src/test/java/JsonParserTests.java]: Do not use plurals

File: src/test/java/JsonParserTests.java [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+        @ValueSource(strings = {"alexey-cart", "111BBvv6", "!@#$%^&"})
+        void writeToFileNotEmptyFilePositive(String cartName) {
+            String fileName = cartName + ".json";
+            cart = new Cart(cartName);
+
+            jsonParser.writeToFile(cart);
+
+            Assertions.assertTrue(isFileNotEmpty(fileName), "File is empty!");
+        }
+
```
Comment (EugenBorisik) [inline src/test/java/JsonParserTests.java]: Remove 74-75 lines

File: src/test/java/JsonParserTests.java [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+
+    }
+
+    @Nested
+    class readFromFileTests {
```
Comment (EugenBorisik) [inline src/test/java/JsonParserTests.java]: Do not use plurals

File: src/test/java/JsonParserTests.java [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+
+        @ParameterizedTest
+        @CsvSource({"src/test/resources/andrew-cart.json, igor-cart, andrew-cart, 38445.479999999996", "src/main/resources/eugen-cart.json, igor-cart, eugen-cart, 26560.68"})
+        void readFromFilePositive(String path, String initCartName, String cartName, double totalPrice) {
+            cart = new Cart(initCartName);
+            String firstCartName = cart.getCartName();
+            double firsTotalPrice = cart.getTotalPrice();
+
+            cart = jsonParser.readFromFile(new File(path));
+
```
Comment (EugenBorisik) [inline src/test/java/JsonParserTests.java]: Remove 1 line

File: src/test/java/JsonParserTests.java [hunk 16]
```diff
@@ -0,0 +1,131 @@
...
+                    () -> Assertions.assertNotEquals(firsTotalPrice, cart.getTotalPrice()),
+                    () -> Assertions.assertEquals(totalPrice, cart.getTotalPrice())
+            );
+
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"src/test/resources/text.csv", "src/main/resources/andrew-cart.jso", "eugen-cart", "src/test/resources/map.txt", "src/main/resources/eugen-cart"})
+    void readFromFileIncorrectTypeExceptionTests(String fileName) {
```
Comment (EugenBorisik) [inline src/test/java/JsonParserTests.java]: Do not use plurals

File: src/test/java/JsonParserTests.java [hunk 17]
```diff
@@ -0,0 +1,131 @@
...
+        File file = new File(fileName);
+        Throwable throwable = Assertions.assertThrows(parser.NoSuchFileException.class, () ->
+                jsonParser.readFromFile(file));
+        Assertions.assertEquals(throwable.getMessage(), String.format("File %s.json not found!", file));
+    }
+
+
+    public boolean isFilePresent(String name) {
+        File file = new File("src/main/resources/" + name);
+        if (file.exists()) {
```
Comment (EugenBorisik) [inline src/test/java/JsonParserTests.java]: Just return file.exists()

File: src/test/java/JsonParserTests.java [hunk 18]
```diff
@@ -0,0 +1,131 @@
...
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public boolean isFileNotEmpty(String name) {
+        File file = new File("src/main/resources/" + name);
+        if (file.length() > 0) {
```
Comment (EugenBorisik) [inline src/test/java/JsonParserTests.java]: Just return file.length() > 0

File: src/test/java/JsonParserTests.java [hunk 19]
```diff
@@ -0,0 +1,131 @@
...
+            return true;
+        } else {
+            return false;
+        }
+    }
+
```
Comment (EugenBorisik) [inline src/test/java/JsonParserTests.java]: Remove 129-130 lines

File: src/test/java/RealItemTests.java [hunk 20]
```diff
@@ -0,0 +1,22 @@
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import shop.RealItem;
+
+public class RealItemTests {
```
Comment (EugenBorisik) [inline src/test/java/RealItemTests.java]: Do not use plurals

File: src/test/java/RealItemTests.java [hunk 21]
```diff
@@ -0,0 +1,22 @@
...
+
```
Comment (EugenBorisik) [inline src/test/java/RealItemTests.java]: Remove 1 line

File: src/test/java/RealItemTests.java [hunk 22]
```diff
@@ -0,0 +1,22 @@
...
+
+    private RealItem realItem;
+
```
Comment (EugenBorisik) [inline src/test/java/RealItemTests.java]: Remove 1 line

File: src/test/java/VirtualItemTests.java [hunk 23]
```diff
@@ -0,0 +1,25 @@
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import shop.VirtualItem;
+
+public class VirtualItemTests {
```
Comment (EugenBorisik) [inline src/test/java/VirtualItemTests.java]: Do not use plurals

File: src/test/java/VirtualItemTests.java [hunk 24]
```diff
@@ -0,0 +1,25 @@
...
+
```
Comment (EugenBorisik) [inline src/test/java/VirtualItemTests.java]: Remove 1 line

File: src/test/java/VirtualItemTests.java [hunk 25]
```diff
@@ -0,0 +1,25 @@
...
+
+    private VirtualItem virtualItem;
+
```
Comment (EugenBorisik) [inline src/test/java/VirtualItemTests.java]: Remove 1 line

File: src/test/java/VirtualItemTests.java [hunk 26]
```diff
@@ -0,0 +1,25 @@
...
+            "Class: class shop.VirtualItem; Name: Microsoft office; Price: 30.0; Size on disk: 8500.0, Microsoft office, 30.0, 8500.0"})
+    void virtualItemToStringMethodTest(String rightValue, String name, double price, double size) {
+        virtualItem = new VirtualItem();
+
+        virtualItem.setName(name);
+        virtualItem.setPrice(price);
+        virtualItem.setSizeOnDisk(size);
+
+        Assertions.assertEquals(rightValue, virtualItem.toString());
+
```
Comment (EugenBorisik) [inline src/test/java/VirtualItemTests.java]: Remove 1 line

**Review by EugenBorisik — APPROVED** [review]

File: src/test/java/JsonParserTests.java (hunk 11 above)
Comment (alex-slapoguzov) [reply src/test/java/JsonParserTests.java]: file = new File("src/main/resources/");
        for (File item : file.listFiles()) {
            if (!(item.getName().equals("andrew-cart.json") | item.getName().equals("eugen-cart.json"))) {
                FileUtils.deleteQuietly(item);
            }

**Pull request merged**